pkg/server/sandbox_run.go

```diff
@@ -179,32 +154,29 @@ func (c *criContainerdService) RunPodSandbox(ctx context.Context, r *runtime.Run
 		}
 	}()
 
-	createOpts := &tasks.CreateTaskRequest{
-		ContainerID: id,
-		Rootfs:      rootfs,
-		// No stdin for sandbox container.
-		Stdout: stdout,
-		Stderr: stderr,
+	fifo := &containerd.FIFOSet{
+		Dir: sandboxRootDir,
+		Out: stdout,
+		Err: stderr,
 	}
-	// Create sandbox task in containerd.
-	glog.V(5).Infof("Create sandbox container (id=%q, name=%q) with options %+v.",
-		id, name, createOpts)
-	createResp, err := c.taskService.Create(ctx, createOpts)
+	task, err := container.NewTask(ctx, containerd.NewIOWithFifoSet(os.Stdin, os.Stdout, os.Stderr, fifo, false))
 	if err != nil {
-		return nil, fmt.Errorf("failed to create sandbox container %q: %v",
-			id, err)
+		return nil, fmt.Errorf("failed to create task for sandbox:%q: %v", container.ID(), err)
 	}
 	defer func() {
 		if retErr != nil {
-			// Cleanup the sandbox container if an error is returned.
-			if err := c.stopSandboxContainer(ctx, id); err != nil {
+			if _, err := task.Delete(ctx); err != nil {
```

Comment:
For this PR to work we anyways need to bump containerd version. I can incorporate the change. WDYT ?